lindera/src/character_filter/mapping.rs

```diff
@@ -0,0 +1,188 @@
+use std::collections::HashMap;
+
+use serde::{Deserialize, Serialize};
+use yada::{builder::DoubleArrayBuilder, DoubleArray};
+
+use lindera_core::character_filter::CharacterFilter;
+
+use crate::{error::LinderaErrorKind, LinderaResult};
+
+pub const MAPPING_CHARACTER_FILTER_NAME: &str = "mapping";
+
+#[derive(Clone, Debug, Deserialize, Serialize, PartialEq, Eq)]
+pub struct MappingCharacterFilterConfig {
+    pub mapping: HashMap<String, String>,
+}
+
+impl MappingCharacterFilterConfig {
+    pub fn new(map: HashMap<String, String>) -> Self {
+        Self { mapping: map }
+    }
+
+    pub fn from_slice(data: &[u8]) -> LinderaResult<Self> {
+        serde_json::from_slice(data).map_err(|err| LinderaErrorKind::Deserialize.with_error(err))
+    }
+}
+
+#[derive(Clone)]
+pub struct MappingCharacterFilter {
+    config: MappingCharacterFilterConfig,
+    trie: DoubleArray<Vec<u8>>,
```

Comment:
Is the mapping so large? To me an ordinal HashMap seems to be sufficient (and fast) here.

Reply:
OK, I found Lucene's MappingCharFilter uses FST.